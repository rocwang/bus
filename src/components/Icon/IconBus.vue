<template>
  <path
    fill="currentColor"
    d="M28.594 7.5h-.469V4.687C28.125 2.063 22.312 0 15 0 7.687 0 1.875 2.063 1.875 4.688V7.5h-.469C.63 7.5 0 8.13 0 8.906v4.688C0 14.37.63 15 1.406 15h.469v9.375c0 1.035.84 1.875 1.875 1.875v1.875C3.75 29.16 4.59 30 5.625 30H7.5c1.035 0 1.875-.84 1.875-1.875V26.25h11.25v1.875c0 1.035.84 1.875 1.875 1.875h1.875c1.035 0 1.875-.84 1.875-1.875V26.25h.375c.938 0 1.5-.75 1.5-1.5V15h.469C29.37 15 30 14.37 30 13.594V8.906c0-.777-.63-1.406-1.406-1.406zM6.562 23.438a1.875 1.875 0 1 1 .001-3.751 1.875 1.875 0 0 1 0 3.75zm.938-6.563A1.875 1.875 0 0 1 5.625 15V7.5c0-1.035.84-1.875 1.875-1.875h15c1.035 0 1.875.84 1.875 1.875V15c0 1.035-.84 1.875-1.875 1.875h-15zm15.938 6.563a1.875 1.875 0 1 1 0-3.751 1.875 1.875 0 0 1 0 3.75z"
  />
</template>

<script>
export default {
  name: "IconBus"
};
</script>
