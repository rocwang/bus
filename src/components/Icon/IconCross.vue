<template>
  <path
    fill="none"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    d="M15 15L29 1 15 15 1 1l14 14zm0 0L1 29l14-14 14 14-14-14z"
  />
</template>

<script>
export default {
  name: "IconCross",
};
</script>
