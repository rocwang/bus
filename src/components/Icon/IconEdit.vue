<template>
  <path
    fill="currentColor"
    d="M20.969 4.875l4.698 5.285a.625.625 0 0 1 0 .809L14.292 23.766l-4.834.603c-.646.082-1.192-.533-1.12-1.26l.537-5.437L20.25 4.875a.47.47 0 0 1 .719 0zm8.437-1.342L26.865.673c-.792-.89-2.079-.89-2.875 0l-1.844 2.075a.625.625 0 0 0 0 .809l4.698 5.285a.47.47 0 0 0 .718 0l1.844-2.074c.792-.897.792-2.344 0-3.235zM20 20.285v5.965H3.333V7.5h11.97a.606.606 0 0 0 .442-.205l2.083-2.344c.396-.445.115-1.201-.443-1.201H2.5C1.12 3.75 0 5.01 0 6.563v20.625C0 28.74 1.12 30 2.5 30h18.333c1.38 0 2.5-1.26 2.5-2.813v-9.246c0-.627-.672-.937-1.067-.498l-2.084 2.344a.77.77 0 0 0-.182.498z"
  />
</template>

<script>
export default {
  name: "IconEdit"
};
</script>
