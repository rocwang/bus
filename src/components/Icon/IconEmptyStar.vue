<template>
  <path
    fill="currentColor"
    d="M28.464 10.048L20.27 8.8l-3.662-7.756a1.763 1.763 0 0 0-3.218 0L9.729 8.8l-8.193 1.248c-1.469.223-2.057 2.115-.992 3.2l5.927 6.034-1.402 8.524c-.252 1.54 1.301 2.695 2.602 1.974L15 25.755l7.329 4.025c1.3.715 2.854-.433 2.602-1.974l-1.402-8.524 5.927-6.035c1.065-1.084.477-2.976-.992-3.199zm-7.9 7.994L21.876 26 15 22.245 8.125 26l1.31-7.958-5.564-5.635 7.689-1.162L15 4l3.44 7.245 7.689 1.162-5.564 5.635z"
  />
</template>

<script>
export default {
  name: "IconEmptyStar",
};
</script>
