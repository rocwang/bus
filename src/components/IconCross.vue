<template>
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 30 30">
    <path
      fill="none"
      :stroke="color"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      d="M15 15L29 1 15 15 1 1l14 14zm0 0L1 29l14-14 14 14-14-14z"
    />
  </svg>
</template>

<script>
export default {
  name: "IconCross",
  inject: ["colors"],
  props: {
    color: {
      type: String,
      default() {
        return this.colors.yellow;
      }
    }
  }
};
</script>
