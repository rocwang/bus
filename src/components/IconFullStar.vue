<template>
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 30 30">
    <path
      :fill="color"
      d="M13.39 1.044L9.73 8.8l-8.193 1.248c-1.469.223-2.057 2.115-.992 3.2l5.927 6.034-1.402 8.524c-.252 1.54 1.301 2.695 2.602 1.974L15 25.755l7.329 4.025c1.3.715 2.854-.433 2.602-1.974l-1.402-8.524 5.927-6.035c1.065-1.084.477-2.976-.992-3.199L20.27 8.8l-3.662-7.756a1.763 1.763 0 0 0-3.218 0z"
    />
  </svg>
</template>

<script>
export default {
  name: "IconFullStar",
  inject: ["colors"],
  props: {
    color: {
      type: String,
      default() {
        return this.colors ? this.colors.yellow : "#ffdb1b";
      }
    }
  }
};
</script>
