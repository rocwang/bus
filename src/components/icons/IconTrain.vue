<template>
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 30 30">
    <path
      :fill="color"
      d="M28.25 5.625v15c0 3.036-3.61 5.625-7.618 5.625l3.69 2.913a.47.47 0 0 1-.29.837H6.218a.47.47 0 0 1-.29-.837l3.69-2.913C5.621 26.25 2 23.67 2 20.625v-15C2 2.518 5.75 0 9.5 0h11.25c3.809 0 7.5 2.518 7.5 5.625zm-2.813 7.969V7.03c0-.776-.63-1.406-1.406-1.406H6.22c-.777 0-1.407.63-1.407 1.406v6.563c0 .776.63 1.406 1.407 1.406H24.03c.777 0 1.407-.63 1.407-1.406zm-10.312 3.75a3.281 3.281 0 1 0 0 6.562 3.281 3.281 0 0 0 0-6.562z"
    />
  </svg>
</template>

<script>
export default {
  name: "IconTrain",
  inject: ["colors"],
  props: {
    color: {
      type: String,
      default() {
        return this.colors ? this.colors.oceanBlue : "#001120";
      }
    }
  }
};
</script>
