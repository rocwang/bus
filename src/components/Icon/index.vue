<template>
  <svg
    xmlns="http://www.w3.org/2000/svg"
    viewBox="0 0 30 30"
    :class="$style[direction]"
    :style="{ color }"
  >
    <component :is="pathComponent" />
  </svg>
</template>

<script>
import IconArrow from "./IconArrow";
import IconBus from "./IconBus";
import IconCross from "./IconCross";
import IconEdit from "./IconEdit";
import IconEmptyStar from "./IconEmptyStar";
import IconFullStar from "./IconFullStar";
import IconShip from "./IconShip";
import IconTrain from "./IconTrain";

export default {
  name: "Icon",
  components: {
    IconArrow,
    IconBus,
    IconCross,
    IconEdit,
    IconEmptyStar,
    IconFullStar,
    IconShip,
    IconTrain
  },
  props: {
    name: {
      type: String,
      required: true,
      validator: function(name) {
        return [
          "Arrow",
          "Bus",
          "Cross",
          "Edit",
          "EmptyStar",
          "FullStar",
          "Ship",
          "Train"
        ].includes(name);
      }
    },
    direction: {
      type: String,
      default: "up",
      validator(value) {
        return ["up", "down", "left", "right"].includes(value);
      }
    },
    color: {
      type: String,
      default: "#000"
    }
  },
  computed: {
    pathComponent() {
      return `Icon${this.name}`;
    }
  }
};
</script>

<style module>
.right {
  transform: rotate(90deg);
}

.down {
  transform: rotate(180deg);
}

.left {
  transform: rotate(-90deg);
}
</style>
