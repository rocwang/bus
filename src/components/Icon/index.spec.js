import { mount } from "@vue/test-utils";
import Icon from "./index";
import IconArrow from "./IconArrow";
import IconBus from "./IconBus";
import IconCross from "./IconCross";
import IconEdit from "./IconEdit";
import IconEmptyStar from "./IconEmptyStar";
import IconFullStar from "./IconFullStar";
import IconShip from "./IconShip";
import IconTrain from "./IconTrain";

describe("Icon.vue", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(Icon, {
      propsData: {
        color: "red",
        direction: "down",
        name: "Arrow"
      }
    });
  });

  it("matches the snapshot", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("has a color", () => {
    expect(wrapper.attributes("style")).toMatch(/color:.*/);
  });

  it("has a direction", () => {
    expect(wrapper.classes()).toContain(wrapper.vm.$style.down);
  });

  it("has path components that match the snapshots", () => {
    [
      IconArrow,
      IconBus,
      IconCross,
      IconEdit,
      IconEmptyStar,
      IconFullStar,
      IconShip,
      IconTrain
    ].forEach(pathComponents => {
      expect(mount(pathComponents).element).toMatchSnapshot(
        pathComponents.name
      );
    });
  });
});
