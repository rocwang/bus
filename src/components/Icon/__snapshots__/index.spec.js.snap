// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon.vue has path components that match the snapshots: IconArrow 1`] = `
<path
  d="M2.01 17.245L.484 15.758a1.572 1.572 0 0 1 0-2.27L13.835.472c.646-.63 1.69-.63 2.33 0l13.35 13.01c.646.63.646 1.647 0 2.27l-1.525 1.486a1.681 1.681 0 0 1-2.357-.027L17.752 9.15v19.243c0 .89-.735 1.607-1.65 1.607h-2.198c-.914 0-1.65-.716-1.65-1.607V9.15l-7.887 8.068a1.67 1.67 0 0 1-2.357.027z"
  fill="currentColor"
/>
`;

exports[`Icon.vue has path components that match the snapshots: IconBus 1`] = `
<path
  d="M28.594 7.5h-.469V4.687C28.125 2.063 22.312 0 15 0 7.687 0 1.875 2.063 1.875 4.688V7.5h-.469C.63 7.5 0 8.13 0 8.906v4.688C0 14.37.63 15 1.406 15h.469v9.375c0 1.035.84 1.875 1.875 1.875v1.875C3.75 29.16 4.59 30 5.625 30H7.5c1.035 0 1.875-.84 1.875-1.875V26.25h11.25v1.875c0 1.035.84 1.875 1.875 1.875h1.875c1.035 0 1.875-.84 1.875-1.875V26.25h.375c.938 0 1.5-.75 1.5-1.5V15h.469C29.37 15 30 14.37 30 13.594V8.906c0-.777-.63-1.406-1.406-1.406zM6.562 23.438a1.875 1.875 0 1 1 .001-3.751 1.875 1.875 0 0 1 0 3.75zm.938-6.563A1.875 1.875 0 0 1 5.625 15V7.5c0-1.035.84-1.875 1.875-1.875h15c1.035 0 1.875.84 1.875 1.875V15c0 1.035-.84 1.875-1.875 1.875h-15zm15.938 6.563a1.875 1.875 0 1 1 0-3.751 1.875 1.875 0 0 1 0 3.75z"
  fill="currentColor"
/>
`;

exports[`Icon.vue has path components that match the snapshots: IconCross 1`] = `
<path
  d="M15 15L29 1 15 15 1 1l14 14zm0 0L1 29l14-14 14 14-14-14z"
  fill="none"
  stroke="currentColor"
  stroke-linecap="round"
  stroke-linejoin="round"
  stroke-width="2"
/>
`;

exports[`Icon.vue has path components that match the snapshots: IconEdit 1`] = `
<path
  d="M20.969 4.875l4.698 5.285a.625.625 0 0 1 0 .809L14.292 23.766l-4.834.603c-.646.082-1.192-.533-1.12-1.26l.537-5.437L20.25 4.875a.47.47 0 0 1 .719 0zm8.437-1.342L26.865.673c-.792-.89-2.079-.89-2.875 0l-1.844 2.075a.625.625 0 0 0 0 .809l4.698 5.285a.47.47 0 0 0 .718 0l1.844-2.074c.792-.897.792-2.344 0-3.235zM20 20.285v5.965H3.333V7.5h11.97a.606.606 0 0 0 .442-.205l2.083-2.344c.396-.445.115-1.201-.443-1.201H2.5C1.12 3.75 0 5.01 0 6.563v20.625C0 28.74 1.12 30 2.5 30h18.333c1.38 0 2.5-1.26 2.5-2.813v-9.246c0-.627-.672-.937-1.067-.498l-2.084 2.344a.77.77 0 0 0-.182.498z"
  fill="currentColor"
/>
`;

exports[`Icon.vue has path components that match the snapshots: IconEmptyStar 1`] = `
<path
  d="M28.464 10.048L20.27 8.8l-3.662-7.756a1.763 1.763 0 0 0-3.218 0L9.729 8.8l-8.193 1.248c-1.469.223-2.057 2.115-.992 3.2l5.927 6.034-1.402 8.524c-.252 1.54 1.301 2.695 2.602 1.974L15 25.755l7.329 4.025c1.3.715 2.854-.433 2.602-1.974l-1.402-8.524 5.927-6.035c1.065-1.084.477-2.976-.992-3.199zm-7.9 7.994L21.876 26 15 22.245 8.125 26l1.31-7.958-5.564-5.635 7.689-1.162L15 4l3.44 7.245 7.689 1.162-5.564 5.635z"
  fill="currentColor"
/>
`;

exports[`Icon.vue has path components that match the snapshots: IconFullStar 1`] = `
<path
  d="M13.39 1.044L9.73 8.8l-8.193 1.248c-1.469.223-2.057 2.115-.992 3.2l5.927 6.034-1.402 8.524c-.252 1.54 1.301 2.695 2.602 1.974L15 25.755l7.329 4.025c1.3.715 2.854-.433 2.602-1.974l-1.402-8.524 5.927-6.035c1.065-1.084.477-2.976-.992-3.199L20.27 8.8l-3.662-7.756a1.763 1.763 0 0 0-3.218 0z"
  fill="currentColor"
/>
`;

exports[`Icon.vue has path components that match the snapshots: IconShip 1`] = `
<path
  d="M23.279 20.467l3.282-3.281a1.5 1.5 0 0 0-.602-2.489L24 14.067V7.5A1.5 1.5 0 0 0 22.5 6h-3V4.125C19.5 3.504 18.996 3 18.375 3h-6.75c-.621 0-1.125.504-1.125 1.125V6h-3A1.5 1.5 0 0 0 6 7.5v6.567l-1.959.63a1.5 1.5 0 0 0-.602 2.489l3.282 3.281C5.898 22.513 4.011 24 1.125 24 .504 24 0 24.504 0 25.125v.75C0 26.496.504 27 1.125 27c2.86 0 5.039-.966 6.715-2.784A4.507 4.507 0 0 0 12 27h6a4.507 4.507 0 0 0 4.16-2.784C23.836 26.034 26.014 27 28.875 27c.621 0 1.125-.504 1.125-1.125v-.75c0-.621-.504-1.125-1.125-1.125-2.85 0-4.76-1.453-5.596-3.533zM9 9h12v4.103l-5.541-1.781a1.5 1.5 0 0 0-.918 0L9 13.103V9z"
  fill="currentColor"
/>
`;

exports[`Icon.vue has path components that match the snapshots: IconTrain 1`] = `
<path
  d="M28.25 5.625v15c0 3.036-3.61 5.625-7.618 5.625l3.69 2.913a.47.47 0 0 1-.29.837H6.218a.47.47 0 0 1-.29-.837l3.69-2.913C5.621 26.25 2 23.67 2 20.625v-15C2 2.518 5.75 0 9.5 0h11.25c3.809 0 7.5 2.518 7.5 5.625zm-2.813 7.969V7.03c0-.776-.63-1.406-1.406-1.406H6.22c-.777 0-1.407.63-1.407 1.406v6.563c0 .776.63 1.406 1.407 1.406H24.03c.777 0 1.407-.63 1.407-1.406zm-10.312 3.75a3.281 3.281 0 1 0 0 6.562 3.281 3.281 0 0 0 0-6.562z"
  fill="currentColor"
/>
`;

exports[`Icon.vue matches the snapshot 1`] = `
<svg
  class="down"
  style="color: red;"
  viewBox="0 0 30 30"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M2.01 17.245L.484 15.758a1.572 1.572 0 0 1 0-2.27L13.835.472c.646-.63 1.69-.63 2.33 0l13.35 13.01c.646.63.646 1.647 0 2.27l-1.525 1.486a1.681 1.681 0 0 1-2.357-.027L17.752 9.15v19.243c0 .89-.735 1.607-1.65 1.607h-2.198c-.914 0-1.65-.716-1.65-1.607V9.15l-7.887 8.068a1.67 1.67 0 0 1-2.357.027z"
    fill="currentColor"
  />
</svg>
`;
